### PR TITLE
#106 Call CreateUser Mutation when Signing Up

### LINKED ISSUE #106: Call create user endpoint after user is createed
**Is your feature request related to a problem? Please describe.**
A backend change requires us to now call create user endpoint. 

**Describe the solution you'd like**
Whenever a user is created via email/password, or a successful Google signin occurs, call the create user endpoint with the relevant arguments.

You may need to run the generate script to get the updated endpoint typings.

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### PR DESCRIPTION
**Pull request by frasermcc9**
# Description

Updates the frontend that calls create user after a google signin or email account creation. This is necessary due to backend no longer automatically creating a user on the first request made by that user.

Fixes/resolves #106 

## Type of change

- [X] **Improvement** (non-breaking change which improves existing functionality)

# Checklist:

Leave blank if not applicable

I have completed these steps when making this pull request:

- [x] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [x] I have labelled the PR appropriately
- [x] I have assigned myself to the PR

Before opening the PR for review:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added attributions to new dependencies and resources
- [x] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

**Review by FriedhelmWS — APPROVED** [review]
LGTM

File: packages/backend/src/controllers/house/house.controller.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export class HouseController {
     @User() user: DecodedIdToken,
   ): Promise<HouseResponseDto | null> {
     const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
-    if (userDoc.house != undefined) {
+    if (userDoc?.house != undefined) {
```
Comment (wshe874) [inline packages/backend/src/controllers/house/house.controller.ts:76]: should we consider using strict equality operator, i.e !==

**Review by wshe874 — COMMENTED** [review]
Looks good to me. Just a suggestion in using strict equality operator.

File: packages/backend/src/controllers/house/house.controller.ts (hunk 1 above)
Comment (frasermcc9) [reply packages/backend/src/controllers/house/house.controller.ts:76]: Not sure I just left it the same. I imagine it's just to cover checking both null and undefined.

Comment (j-chad) [reply packages/backend/src/controllers/house/house.controller.ts:76]: yes that is why it's there

**Review by uoa-aron527 — APPROVED** [review]
Looks good to me! :-)

**Review by j-chad — APPROVED** [review]
LGTM

**Pull request merged**